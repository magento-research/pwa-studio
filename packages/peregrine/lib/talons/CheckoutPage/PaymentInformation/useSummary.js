import { useQuery } from '@apollo/react-hooks';

import { useCartContext } from '../../../context/cart';

/**
 * Talon to handle summary component in payment information section of
 * the checkout page.
 *
 * @param {DocumentNode} props.operations.queries.getBillingAddressQuery query to get saved billing address from cache
 * @param {DocumentNode} props.operations.queries.getIsBillingAddressSameQuery query to get if billing address is same as shipping address from cache
 *
 * @returns {
 *   billingAddress: {
 *      firstName: String,
 *      lastName: String,
 *      country: String,
 *      street1: String,
 *      street2: String,
 *      city: String,
 *      state: String,
 *      postalCode: String,
 *   },
 *   isBillingAddressSame: Boolean
 * }
 */
const useSummary = props => {
    const { operations } = props;
    const {
        queries: { getBillingAddressQuery, getIsBillingAddressSameQuery }
    } = operations;

    /**
     * Definitions
     */

    const [{ cartId }] = useCartContext();

    /**
     * Queries
     */

    const { data: billingAddressData } = useQuery(getBillingAddressQuery, {
        variables: { cartId }
    });

    const { data: isBillingAddressSameData } = useQuery(
        getIsBillingAddressSameQuery,
        { variables: { cartId } }
    );

    const billingAddress = billingAddressData
        ? billingAddressData.cart.billingAddress
        : {};

    const isBillingAddressSame = isBillingAddressSameData
        ? isBillingAddressSameData.cart.isBillingAddressSame
        : true;

    return {
        billingAddress,
        isBillingAddressSame
    };
};

export default useSummary;
