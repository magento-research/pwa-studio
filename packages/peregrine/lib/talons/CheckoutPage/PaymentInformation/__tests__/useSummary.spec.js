import React from 'react';

import { useSummary } from '../useSummary';
import createTestInstance from '../../../../util/createTestInstance';

jest.mock('../../../../context/cart', () => ({
    useCartContext: jest.fn().mockReturnValue([{ cartId: '123' }])
}));

jest.mock('@apollo/react-hooks', () => ({
    useQuery: jest
        .fn()
        .mockReturnValueOnce({
            data: {
                cart: {
                    billingAddress: {
                        firstName: 'Goosey',
                        lastName: 'Goose',
                        country: 'United States of Gooseland',
                        street1: '12345 Gooseey Blvd',
                        street2: 'Apt 123',
                        city: 'Goostin',
                        state: 'Gooseyork',
                        postalCode: '12345',
                        phoneNumber: '1234567890'
                    }
                }
            }
        })
        .mockReturnValueOnce({
            data: {
                cart: {
                    isBillingAddressSame: false
                }
            }
        })
        .mockReturnValue()
}));

const Component = props => {
    const talonProps = useSummary(props);

    return <i talonProps={talonProps} />;
};

test('Snapshot test', () => {
    const tree = createTestInstance(<Component operations={{ queries: {} }} />);
    const { root } = tree;
    const { talonProps } = root.findByType('i').props;

    expect(talonProps).toMatchSnapshot();
});
