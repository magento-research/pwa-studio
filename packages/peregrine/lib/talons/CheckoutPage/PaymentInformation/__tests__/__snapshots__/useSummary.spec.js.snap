// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "billingAddress": Object {
    "city": "Goostin",
    "country": "United States of Gooseland",
    "firstName": "Goosey",
    "lastName": "Goose",
    "phoneNumber": "1234567890",
    "postalCode": "12345",
    "state": "Gooseyork",
    "street1": "12345 Gooseey Blvd",
    "street2": "Apt 123",
  },
  "isBillingAddressSame": false,
}
`;
